flow/src/main/java/com/sequenceiq/flow/repository/FlowChainLogRepository.java

```diff
@@ -28,4 +31,14 @@ public interface FlowChainLogRepository extends CrudRepository<FlowChainLog, Lon
             + "WHERE fch.flowChainId NOT IN ( SELECT DISTINCT fl.flowChainId FROM FlowLog fl )"
             + " AND fch.flowChainId NOT IN (SELECT DISTINCT fc.parentFlowChainId FROM FlowChainLog fc)")
     int purgeOrphanFlowChainLogs();
+
+    @Query(value = "SELECT fcl.* " +
+            "FROM FlowChainLog fcl " +
+            "INNER JOIN ( " +
+            "    SELECT flowChainId, max(created) as created " +
+            "    FROM FlowChainLog " +
+            "    WHERE flowChainId IN (:flowChainIds) " +
+            "    GROUP by flowChainId " +
+            ")fch on fcl.flowChainId = fch.flowChainId and fcl.created = fch.created", nativeQuery = true)
+    List<FlowChainLog> nativeFindByFlowChainIdInOrderByCreatedDesc(@Param("flowChainIds") Set<String> flowChainIds, Pageable pageable);
```

Comment:
I don't think pageable has any effect for a native query without including it in the query itself

Reply:
Pageable has effect in native query as well - https://www.bezkoder.com/spring-data-pageable-custom-query/